src/main/java/eu/darkbot/api/managers/PetManager.java

```diff
@@ -0,0 +1,63 @@
+package eu.darkbot.api.managers;
+
+import eu.darkbot.api.entities.Pet;
+import eu.darkbot.api.objects.Gui;
+import eu.darkbot.utils.ItemNotEquippedException;
+
+public interface PetManager extends Gui, Pet {
+
+    boolean isGearAvailable(int gearId);
+
+    default boolean isGearAvailable(Pet.Gear gear) {
+        return isGearAvailable(gear.getId());
+    }
+
+    void setGear(int gearId) throws ItemNotEquippedException;
+
+    default void setGear(Pet.Gear gear) throws ItemNotEquippedException {
+        setGear(gear.getId());
+    }
+
+    boolean hasCooldown(int cooldownId);
+
+    default boolean hasCooldown(Pet.Cooldown cooldown) {
+        return hasCooldown(cooldown.getId());
+    }
+
+    boolean isEnabled();
+    void setEnabled(boolean enabled);
+
+    boolean isActive();
+    boolean isRepaired();
+
+    int getFuel();
+    int getMaxFuel();
+
+    int getHeat();
+    int getMaxHeat();
+
+    @Override
+    default int getHull() {
+        throw new UnsupportedOperationException("Pet dont have nano hull");
```

Comment:
```suggestion
        throw new UnsupportedOperationException("Pet has no nano hull");
```
consider it may be simpler to return 0 than to throw an exception, simply pretending a pet *does* have a hull, it just turns out to be of always 0, max 0, and never increases nor decreases